packages/contracts/contracts/engine/V3/EngineFactoryV0.sol

```diff
@@ -0,0 +1,215 @@
+        // set owner of core registry contract
+        Ownable(coreRegistry_).transferOwnership(address(this));
+        // emit event
+        emit Deployed({
+            engineImplementation: engineImplementation_,
+            engineFlexImplementation: engineFlexImplementation_,
+            type_: type_
+        });
+    }
+
+    /**
+     * @notice Creates a new Engine or Engine Flex contract with the provided
+     * `engineConfiguration`, depending on the `engineCoreType`.
+     * @param engineCoreType Type of Engine Core contract.
+     * @param engineConfiguration EngineConfiguration data to configure the
+     * contract with.
+     * @param adminACLContract Address of admin access control contract, to be
+     * set as contract owner. A new contract will be deployed if address is null.
+     * @return engineContract The address of the newly created Engine or Engine Flex
+     * contract. The address is also emitted in both the `EngineCreated` and
+     * `EngineFlexCreated` events.
+     */
+    function createEngineContract(
+        IEngineFactoryV0.EngineCoreType engineCoreType,
+        EngineConfiguration calldata engineConfiguration,
+        address adminACLContract
+    ) external onlyOwner returns (address engineContract) {
+        require(!isAbandoned, "factory is abandoned");
+        // validate engine contract configuration
+        _onlyNonZeroAddress(engineConfiguration.renderProviderAddress);
+        _onlyNonZeroAddress(engineConfiguration.randomizerContract);
+        if (adminACLContract == address(0)) {
+            // deploy new AdminACLV0 contract
+            adminACLContract = Create2.deploy({
+                amount: 0,
+                salt: keccak256(abi.encodePacked(msg.sender, block.timestamp)),
+                bytecode: type(AdminACLV0).creationCode
+            });
+        }
+
+        address implementation = engineCoreType ==
+            IEngineFactoryV0.EngineCoreType.Engine
+            ? engineImplementation
+            : engineFlexImplementation;
+
+        engineContract = Clones.clone({implementation: implementation});
+
+        IGenArt721CoreContractV3_Engine(engineContract).initialize(
+            engineConfiguration,
+            adminACLContract
+        );
+
+        // update super admin address if not null address
+        if (engineConfiguration.newSuperAdminAddress != address(0)) {
+            address[] memory genArt721CoreAddressesToUpdate = new address[](1);
+            genArt721CoreAddressesToUpdate[0] = engineContract;
+
+            IAdminACLV0(adminACLContract).changeSuperAdmin(
+                engineConfiguration.newSuperAdminAddress,
+                genArt721CoreAddressesToUpdate
+            );
+        }
+
+        string memory coreType = IGenArt721CoreContractV3_Engine(engineContract)
+            .coreType();
+        string memory coreVersion = IGenArt721CoreContractV3_Engine(
+            engineContract
+        ).coreVersion();
```

Comment:
nit: these can be queried and stored in local state variables in this contract's constructor, since they will be the same across all deployments (one pair for engine, one pair for engine flex)